nodejs/query/sources.ts

```diff
@@ -47,7 +47,9 @@ export function range(start: number, end?: number): AsyncIterable<number> {
 }
 
 export function from<TSource>(source: AsyncQuerySource<TSource>): AsyncIterable<TSource> {
-    return source instanceof Function
+    return source === undefined
```

Comment:
Should we handle any falsy value in the same way (e.g. `null`)?

Reply:
I actually meant to protect specifically against spurious `undefined` (via accident-prone things like `!`) vs. `null` which I think tends to be intentional—you have to cast it to the type, I think, to pass it? We can do `null` too though

Reply:
What check did you have in mind for all falsey values?

Reply:
> I actually meant to protect specifically against spurious `undefined` (via accident-prone things like `!`) vs. `null` which I think tends to be intentional—you have to cast it to the type, I think, to pass it? We can do `null` too though

From JavaScript you could pass anything...

> What check did you have in mind for all falsey values?

```typescript
return !source ? wrapAsyncIterable(() => []) : ...
```

I could also see throwing an error.